bbid.py

```diff
@@ -94,7 +100,8 @@ def backup_history(*args):
 	download_history=open(output_dir + '/download_history.pickle','wb')
 	pickle.dump(tried_urls,download_history)
 	pickle.dump(finished_keywords, download_history)
-	pickle.dump(image_md5s, download_history)
+	copied_image_md5s = dict(image_md5s)  # To resolve a concurrency issue
+	pickle.dump(copied_image_md5s, download_history)
```

Comment:
Yes I've encountered an error on this line saying the dictionary has been updated during this.
That's why I fixed this part.
Since backup_history() is a signal handler,
it can be called while other download threads are alive.